### PR TITLE
T/1006 Expanded special case in `Range#_getTransformedByDelta`.

### LINKED ISSUE #1006: Range at the end of element is incorrectly transformed when split
Originally reported here: https://github.com/ckeditor/ckeditor5/issues/485

The issue is that in `Range#_getTransformedByDelta` the range is incorrectly transformed. Two ranges (that are touching) are returned through default algorithm. Then, one of those ranges is ignored. One range should be returned instead.

### PR DESCRIPTION
**Pull request by scofalik**
### Suggested merge commit message ([convention](https://github.com/ckeditor/ckeditor5-design/wiki/Git-commit-message-convention))

Fixed: Live ranges and markers, that are at the end of an element, are now correctly transformed when they are split. Closes #1006.

### CODE REVIEW

File: src/model/range.js [hunk 1]
```diff
@@ -462,13 +462,13 @@ export default class Range {
...
 
-			// Edge case for merge detla.
+			// Edge case for merge delta.
 			if (
 				deltaType == 'merge' &&
 				this.isCollapsed &&
 				( this.start.isEqual( sourceRange.start ) || this.start.isEqual( sourceRange.end ) )
 			) {
-				// Collapsed range is in merged element.
+				// Collapsed range is in merged element, at the beginning or at the end of it.
```
Comment (Reinmar) [inline src/model/range.js]: Where's the "at the beginning" part checked in the code?

Comment (Reinmar) [inline src/model/range.js]: Actually, the whole condition got too complex and is hard to read. Could you move it to some helper fn and split into a couple of statements?

File: tests/model/range.js [hunk 2]
```diff
@@ -918,8 +918,12 @@ describe( 'Range', () => {
 		} );
 
 		describe( 'by SplitDelta', () => {
-			it( 'split inside range', () => {
+			beforeEach( () => {
+				// Cleaning...
 				root.removeChildren( root.childCount );
```
Comment (Reinmar) [inline tests/model/range.js]: removeChildren() accepts `index + howMany` so this looks like a mistake.

File: src/model/range.js (hunk 1 above)
Comment (scofalik) [inline src/model/range.js]: > Where's the "at the beginning" part checked in the code?

`this.start.isEqual( sourceRange.start )`

Or do you mean tests?

> Actually, the whole condition got too complex and is hard to read.

I haven't changed it. Honestly there's not much to simplify here, except of the last part. I can make some kind of `isAtTheBeginningOrEndOfRange( this.start, sourceRange )` or `isRangeBoundary( this.start, sourceRange )` but IDK how much it helps.

File: tests/model/range.js (hunk 2 above)
Comment (scofalik) [inline tests/model/range.js]: Actually there's no need to add and remove any content from root in those tests...

**Pull request merged**